src/controller.c

```diff
@@ -36,27 +30,30 @@ int is_user_input() {
     return 0;
 }
 
-void act_on_user_input(char user_input, State* s,
+void act_on_user_input(
+    char user_input, 
+    Movement* m,
     int* frame_counter) {
 
     int frames_until_next_move;
     switch(user_input) {
-        case DOWN_KEY: // move block down
-            frames_until_next_move = MOVE_RATE - (*frame_counter % MOVE_RATE);
-            if (frames_until_next_move == 0) frames_until_next_move = MOVE_RATE;
-            *frame_counter += frames_until_next_move;
+        case DOWN_KEY:
+            m->y = 1; // Move down by 1
+            *frame_counter = 0; // Reset counter for default downwards move
             break;
-        case LEFT_KEY: // move block left
-            move_block(s, -1, NO_ROTATE);
+        case LEFT_KEY:
+            m->x += -1; // Move left by 1
             break;
-        case RIGHT_KEY: // move block right
-            move_block(s, 1, NO_ROTATE);
+        case RIGHT_KEY:
+            m->x += 1; // Move right by 1
             break;
         case ROTATE_CW_KEY: // rotate block clockwise
-            move_block(s, 0, RIGHT);
+            if (m->r == NO_ROTATE) {m->r = RIGHT;}
```

Comment:
Could you move these one-liner if statements onto multiple lines?

if (m->r == NO_ROTATE) {
    m->r = RIGHT;
}